compile.sh

```diff
@@ -32,31 +34,48 @@ function gen-swig-java {
 }
 
 function linux {
-    g++ -c -fPIC -I${JAVA_HOME}/include -I${JAVA_HOME}/include/linux src/bindings/java/omega_edit_wrap.cxx -o lib/omega_edit_wrap.o
-    g++ -shared -fPIC -o lib/libomega_edit.so lib/omega_edit_wrap.o -lc
+    includes="-I${PWD}/src/include -I${PWD}/vendor/cwalk/include -I${JAVA_HOME}/include -I${JAVA_HOME}/include/linux"
+    g++ -c -fPIC $includes src/bindings/java/omega_edit_wrap.cxx -o lib/omega_edit_wrap.o
+    g++ -shared -fPIC -o lib/libomega_edit.so lib/omega_edit_wrap.o cmake-build-debug/vendor/cwalk/libcwalk.a -lc
 }
 
 function mac {
-    g++ -std=c++11 -c -fPIC -I${JAVA_HOME}/include -I${JAVA_HOME}/include/darwin src/bindings/java/omega_edit_wrap.cxx -o lib/omega_edit_wrap.o
-    g++ -std=c++11 -dynamiclib -o lib/libomega_edit.dylib cmake-build-debug/libomega_edit.a lib/omega_edit_wrap.o -lc
+    includes="-I${PWD}/src/include -I${PWD}/vendor/cwalk/include -I${JAVA_HOME}/include -I${JAVA_HOME}/include/darwin "
+    g++ -std=c++14 -c -fPIC $includes src/bindings/java/omega_edit_wrap.cxx -o lib/omega_edit_wrap.o
+    g++ -std=c++14 -dynamiclib -o lib/libomega_edit.dylib lib/omega_edit_wrap.o cmake-build-debug/libomega_edit.a cmake-build-debug/vendor/cwalk/libcwalk.a -lc
 }
 
 function win {
-    "g++ -c -I'%JAVA_HOME%/include' -I'%JAVA_HOME%/include/win32' src/bindings/java/omega_edit_wrap.cxx -o lib/omega_edit_wrap.o"
-    "g++ -shared -o lib/libomega_edit.dll cmake-build-debug/libomega_edit.a lib/omega_edit_wrap.o -Wl,--add-stdcall-alias"
+    JAVA_HOME=$(echo $JAVA_HOME | sed -e "s|C:|/c|" | sed -e 's|\\|/|g')
+    g++ -std=gnu++14 -c -I"${PWD}/src/include" -I"${PWD}/vendor/cwalk/include" -I"${JAVA_HOME}/include" -I"${JAVA_HOME}/include/win32" src/bindings/java/omega_edit_wrap.cxx -o lib/omega_edit_wrap.o
+    g++ -shared -o lib/omega_edit.dll lib/omega_edit_wrap.o cmake-build-debug/libomega_edit.a cmake-build-debug/vendor/cwalk/libcwalk.a -Wl,--add-stdcall-alias
+}
+
+function win-ci {
+    cmake -S . -B cmake-build-debug
+    sed -i "s|^CMAKE_CXX_COMPILER:FILEPATH=.*|CMAKE_CXX_COMPILER:FILEPATH=/mingw64/bin/c++|g" cmake-build-debug/CMakeCache.txt
+    cmake --build cmake-build-debug
```

Comment:
Still hate that we have to do this.

Reply:
I have an idea of what could cause the issue I can test but may not work fully will document here